tests/drop_counters/test_configurable_drop_counters.py

```diff
@@ -0,0 +1,214 @@
+
+    Returns: A Dictionary with the following information:
+        "ports": A List of the active ports in the system
+        "phy_intfs": A List of ALL physical interfaces in the system
+        "vlan_intfs": A Dictionary containing all the VLAN interfaces in the system, including
+            their address, subnet, prefix length, and VLAN members.
+    """
+
+    if testbed["topo"]["name"] != "t0":
+        pytest.skip("Unsupported topology {}".format(testbed["topo"]["name"]))
+
+    minigraph_facts = duthost.minigraph_facts(host=duthost.hostname)["ansible_facts"]
+
+    active_ports = natsorted(minigraph_facts["minigraph_ports"].keys())
+
+    physical_interfaces = natsorted(minigraph_facts["minigraph_port_name_to_alias_map"].keys())
+
+    vlan_interfaces = {}
+    for vlan_intf in minigraph_facts["minigraph_vlan_interfaces"]:
+        intf_name = vlan_intf["attachto"]
+        vlan_interfaces[intf_name] = {
+            "addr": vlan_intf["addr"],
+            "subnet": vlan_intf["subnet"],
+            "prefix_len": vlan_intf["prefixlen"],
+            "members": minigraph_facts["minigraph_vlans"][intf_name]["members"]
+        }
+
+    return {
+        "ports": active_ports,
+        "phy_intfs": physical_interfaces,
+        "vlan_intfs": vlan_interfaces
+    }
+
+@pytest.fixture(scope="module")
+def device_capabilities(duthost):
+    """
+    Gathers information about the DUT's drop counter capabilities.
+
+    Returns:
+        A Dictionary of device capabilities (see `get_device_capabilities` under the
+        `configurable_drop_counters` package).
+    """
+
+    capabilities = cdc.get_device_capabilities(duthost)
+
+    if not capabilities:
+        pytest.fail("Error fetching device capabilities")
+
+    return capabilities
+
+@pytest.fixture(params=cdc.SUPPORTED_COUNTER_TYPES)
+def setup_counters(request, device_capabilities, duthost):
+    """
+    Returns a method to setup drop counters.
+
+    Notes:
+        This fixture will automatically clean-up created drop counters.
+
+    Returns:
+        A method which, when called, will create a drop counter with the specified drop reasons.
+    """
+
+    if request.param not in device_capabilities["counters"]:
+        pytest.skip("Counter type not supported on target DUT")
+
+    counter_type = request.param
+    supported_reasons = device_capabilities["reasons"][counter_type]
+
+    def _setup_counters(drop_reasons):
+        if any(reason not in supported_reasons for reason in drop_reasons):
+            pytest.skip("Drop reasons not supported on target DUT")
+
+        cdc.create_drop_counter(duthost, "TEST", counter_type, drop_reasons)
+        time.sleep(1)
+
+        return counter_type
+
+    yield _setup_counters
+
+    try:
+        cdc.delete_drop_counter(duthost, "TEST")
+        time.sleep(1)
+    except Exception:
+        logging.info("Drop counter does not exist, skipping delete step...")
+
+@pytest.fixture
+def test_runner(duthost, ptfadapter, testbed_params):
```

Comment:
Name `test_runner` looks like a test case. Maybe we should avoid using names with prefix `test_` or suffix `_test` if it is not a test case?